toolchain/semantics/semantics_parse_tree_handler.cpp

```diff
@@ -173,20 +306,53 @@ auto SemanticsParseTreeHandler::HandleBreakStatementStart(
 
 auto SemanticsParseTreeHandler::HandleCallExpression(ParseTree::Node parse_node)
     -> bool {
-  emitter_->Emit(parse_node, SemanticsTodo, "HandleCallExpression");
-  return false;
+  auto on_start = [&](SemanticsNodeBlockId ir_id,
+                      SemanticsNodeBlockId refs_id) -> bool {
+    // TODO: Convert to call expression.
+    auto [call_expr_parse_node, name_id] = node_stack_.PopForParseNodeAndNodeId(
+        ParseNodeKind::CallExpressionStart);
+    auto name_node = semantics_->GetNode(name_id);
+    if (name_node.kind() != SemanticsNodeKind::FunctionDeclaration) {
+      // TODO: Work on error.
+      emitter_->Emit(parse_node, SemanticsTodo, "Not a callable name");
+      node_stack_.Push(parse_node, name_id);
+      return true;
+    }
+
+    auto callable_id = name_node.GetAsFunctionDeclaration();
+    auto callable = semantics_->callables_[callable_id.index];
+
+    if (!TryTypeConversionOnArgs(call_expr_parse_node, ir_id, refs_id,
+                                 name_node.parse_node(),
+                                 callable.param_refs_id)) {
+      node_stack_.Push(parse_node, SemanticsNodeId::BuiltinInvalidType);
+      return true;
+    }
+
+    auto call_id = semantics_->AddCall({ir_id, refs_id});
+    // TODO: Propagate return types from callable.
+    auto call_node_id = AddNode(SemanticsNode::MakeCall(
+        call_expr_parse_node, SemanticsNodeId::BuiltinEmptyTuple, call_id,
+        callable_id));
+
+    node_stack_.Push(parse_node, call_node_id);
+    return true;
+  };
+  return ParamOrArgEnd(ParseNodeKind::CallExpressionStart,
+                       ParseNodeKind::CallExpressionComma, on_start);
 }
 
 auto SemanticsParseTreeHandler::HandleCallExpressionComma(
     ParseTree::Node parse_node) -> bool {
-  emitter_->Emit(parse_node, SemanticsTodo, "HandleCallExpressionComma");
-  return false;
+  return ParamOrArgComma(parse_node);
 }
 
 auto SemanticsParseTreeHandler::HandleCallExpressionStart(
     ParseTree::Node parse_node) -> bool {
-  emitter_->Emit(parse_node, SemanticsTodo, "HandleCallExpressionStart");
-  return false;
+  auto name_id = node_stack_.PopForNodeId(ParseNodeKind::NameReference);
```

Comment:
What happens if the callee is some other kind of parse node?

Reply:
Right now it'll just error out on the semantics -- PopForNodeId checks. This is generally how I've been moving along, building minimal support.

Reply:
I see, it looks like this will `CHECK`-fail. This seems like something a fuzzer would be able to find fairly easily (eg, `(1+1)()`), so maybe it's worth adding a temporary "not supported yet" diagnostic? Happy to let you make the call on that; feel free to land as-is if you prefer.

Reply:
I'm going to leave this alone -- while I think the behavior needs to be changed in the future, I don't think it makes sense to change one case when every use of node_stack_ shares the issue.